src/shared/molecules/ResolvedLinkEditorPopover/ResolvedLinkEditorPopover.tsx

```diff
@@ -88,76 +63,39 @@ const PopoverContainer = forwardRef<HTMLDivElement, PopoverContainer>(
 
 const ResolvedLinkEditorPopover = () => {
   const ref = useRef<HTMLDivElement>(null);
-  const navigate = useHistory();
-  const dispatch = useDispatch();
-  const nexus = useNexusContext();
-  const { pathname, search, state } = useLocation();
-  const routeMatch = useRouteMatch<{
-    orgLabel: string;
-    projectLabel: string;
-    resourceId: string;
-  }>(`/:orgLabel/:projectLabel/resources/:resourceId`);
+  const {
+    clickOutsideHandler,
+    navigateResourceHandler,
+    downloadBinaryAsyncHandler,
+  } = useResolvedLinkEditorPopover();
+
   const {
     editorPopoverResolvedData: { open, error, resolvedAs, results },
   } = useSelector((state: RootState) => state.uiSettings);
-  const resultPattern: TResultPattern = { resolvedAs, open };
-
-  const onClickOutside = () => {
-    dispatch({
-      type: UISettingsActionTypes.UPDATE_JSON_EDITOR_POPOVER,
-      payload: editorPopoverResolvedDataInitialValue,
-    });
-  };
-  const onClickLink = async (resource: TDELink) => {
-    onClickOutside();
-    if (pathname === '/data-explorer/graph-flow') {
-      dispatch(AddNewNodeDataExplorerGraphFlow(resource));
-    } else if (routeMatch?.url && routeMatch.params) {
-      const data = (await nexus.Resource.get(
-        routeMatch.params.orgLabel,
-        routeMatch.params.projectLabel,
-        routeMatch.params.resourceId
-      )) as Resource;
-      const orgProject = getOrgAndProjectFromProjectId(data._project);
-      dispatch(
-        InitNewVisitDataExplorerGraphView({
-          referer: { pathname, search, state },
-          source: {
-            _self: data._self,
-            title: getResourceLabel(data),
-            types: getNormalizedTypes(data['@type']),
-            resource: [
-              orgProject?.orgLabel ?? '',
-              orgProject?.projectLabel ?? '',
-              data['@id'],
-            ],
-          },
-          current: resource,
-        })
-      );
-      navigate.push('/data-explorer/graph-flow');
-    }
-  };
 
-  const handleDownloadBinary = async (data: TDELink) => {
-    await downloadFile({
-      nexus,
-      orgLabel: data.resource?.[0]!,
-      projectLabel: data.resource?.[1]!,
-      resourceId: data.resource?.[2]!,
-      ext: data.resource?.[4] ?? 'json',
-      title: data.title,
-    });
-  };
+  const resultPattern: TResultPattern = { resolvedAs, open };
 
   const {
     mutateAsync: downloadBinaryAsync,
     isLoading: downloadInProgress,
-  } = useMutation([], handleDownloadBinary);
+  } = useMutation({
+    mutationKey: 'downloadBinary',
+    mutationFn: downloadBinaryAsyncHandler,
+    onError: error => {
+      notification.error({
+        message: 'Download error',
+        description: 'Something went wrong while downloading the resource.',
+      });
+      Sentry.captureException({
+        error,
+        message: 'Download Binary error',
```

Comment:
It might be helpful to add the url of the resource(s) that the download failed for in Sentry's error message.